### PR TITLE
remove outdated info about seeding in multi-gpu docs 

### LINKED ISSUE #6013: Initialising modules with different weights with DDP is dangerously buried in the doc notes
## 🐛 Bug

More a discussion than a real bug.

The [docs](https://pytorch-lightning.readthedocs.io/en/stable/multi_gpu.html#data-parallel) for the DDP say:

>> Make sure to set the random seed before the instantiation of a Trainer() so that each model initializes with the same weights.

In the most common settings, it is considered bad practice to initialise the modules with different parameters according to the GPU.
To me - and this is a personal opinion - it does not make sense in theory as well.

The reason why I posted it as a bug and not as a doc issue is that this is really **dangerous,** if you don't know about it.
Using a note in the docs does not help either. I don't assume a user reads the entire docs before using a library.

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

We use the pytorch DDP wrapper, which takes care of broadcasting the weights before training starts. 
Therefore, it is completely optional that the user sets a seed when training with DDP.

Fixes #6013

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by SeanNaren — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]
